### PR TITLE
3595 Adds nibabel/pil writers

### PR DESCRIPTION
**Pull request by wyli**
part of #3595 

### Description
adds a Nibabelwriter, PILWriter

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by wyli** [comment]
@Nic-Ma this is ready, mainly replaces `write_nifti` and `write_png`.

File: tests/test_nifti_rw.py [hunk 1]
```diff
@@ -95,9 +95,15 @@ def test_orientation(self, array, affine, reader_param, expected):
...
-            write_nifti(data_array, test_image, header["affine"], header.get("original_affine", None))
+            writer_obj = NibabelWriter()
+            writer_obj.set_data_array(data_array, channel_dim=None)
+            writer_obj.set_metadata(header)
+            writer_obj.write(test_image, verbose=True)
         elif affine is not None:
-            write_nifti(data_array, test_image, affine)
+            writer_obj = NibabelWriter()
+            writer_obj.set_data_array(data_array, channel_dim=None)
+            writer_obj.set_metadata({"affine": affine})
```
Comment (Nic-Ma) [inline tests/test_nifti_rw.py:105]: Seems the `writer_obj` code is duplicated except `set_metadata`?

Thanks.

**Review by Nic-Ma — APPROVED** [review]
The overall design looks good to me.
Could you please help also add the numpy writer to be consistent with readers?
And I think need to double confirm the data shape when `reading` and `writing` with mixed ITK / Nibabel Reader / Writer.

Thanks.

File: docs/source/data.rst [hunk 2]
```diff
@@ -163,6 +163,16 @@ ITKWriter
...
+NibabelWriter
+~~~~~~~~~~~~~
+.. autoclass:: NibabelWriter
+  :members:
+
+PILWriter
+~~~~~~~~~
+.. autoclass:: PILWriter
+  :members:
+
```
[the remark this answers is not in the document]
Comment (wyli) [reply docs/source/data.rst:175]: sure, this is a new feature request, I'll revisit this one after finishing the milestone.

**Pull request merged**